### PR TITLE
fix: get artefact and metafile paths from Serverless API

### LINKED ISSUE #1: cannot find or open .serverless/stack-stage-function, .serverless/stack-stage-function.zip or .serverless/stack-stage-function.ZIP.
It looks as though the plugin is searching for a file named `.serverless/stack-stage-function.zip` or similar, however with my config, `serverless-esbuild` outputs a file named `.serverless/function.zip`

Serverless:
  Framework Core: 2.70.0 (local)
  Plugin: 5.5.3
  SDK: 4.3.0
  Components: 3.18.1

esbuild: 0.14.21

severless-esbuild: 1.24.1

esbuild-visualiser: 0.3.1

serverless-analyze-bundle-plugin: 1.0.5

`serverless.yml`:

```yml
service: my-api
frameworkVersion: '2'
variablesResolutionMode: 20210326

package:
  individually: true

provider:
  name: aws
  runtime: nodejs14.x
  lambdaHashingVersion: 20201221
  stage: ${opt:stage}
  region: eu-west-1
  
plugins:
  - serverless-esbuild
  - serverless-analyze-bundle-plugin

functions:
  foobar:
    description: Foobar function
    handler: src/handlers/foo.main

custom:
  esbuild:
    packager: yarn
    bundle: true
    minify: true
    target: node14
    sourcemap: external
```

### PR DESCRIPTION
**Pull request by smbkr**
Gets the paths to the metadata file and the zipped artifact from the FunctionDefinitionHandler object exposed by the Serverless API. 

The current release fails for me where e.g. the function name in Serverless is 'myFunction' but the file path is `myFunctionHandler.ts`, as the current implementation assumes the filename and function name are the same which may not be the case. ESBuild outputs the metafile as `${sourceFileName}-meta.json`.

The current implementation also assumes the full function name (`stack-stage-function`) is the same as the output zip file, however in my case the output file is just the function name sans stage and stack. This PR fixes this by getting the path from the Serverless API.

Fixes #1

### CODE REVIEW

File: plugin/bundleVisualizer.ts [hunk 1]
```diff
@@ -43,12 +44,10 @@ async function bundleVisualizer(this: ServerlessAnalyzeBundlePlugin): Promise<vo
...
     return;
   }
+  const slsFunction = this.serverless.service.getFunction(functionName);
 
-  const functionZipName = this.serverless.service
-    .getAllFunctionsNames()
-    .find(fullFunctionName => fullFunctionName.endsWith(functionName));
-
-  if (functionZipName === undefined) {
+  const fullZipPath = slsFunction.package?.artifact;
```
Comment (smbkr) [inline plugin/bundleVisualizer.ts:49]: something like `/home/me/projects/foo/.serverless/myfunction.zip`

File: plugin/bundleVisualizer.ts [hunk 2]
```diff
@@ -57,6 +56,7 @@ async function bundleVisualizer(this: ServerlessAnalyzeBundlePlugin): Promise<vo
 
     return;
   }
+  const functionZipName = parse(fullZipPath).base;
```
Comment (smbkr) [inline plugin/bundleVisualizer.ts:59]: just the file and ext, e.g. `myfunction.zip`

File: plugin/bundleVisualizer.ts [hunk 3]
```diff
@@ -65,10 +65,23 @@ async function bundleVisualizer(this: ServerlessAnalyzeBundlePlugin): Promise<vo
 
   await pExec(`unzip .serverless/${functionZipName} -d ${TEMP_DIR_LOCATION}`);
 
-  const test = getAllFiles(`${TEMP_DIR_LOCATION}`);
-  const metafileName = test.filter(
-    fileName => fileName.includes(`/${functionName}/`) && fileName.endsWith('-meta.json'),
-  )[0];
+  const allFiles = getAllFiles(`${TEMP_DIR_LOCATION}`);
+  const metafileName = allFiles.filter(fileName => {
+    const handlerPath = (slsFunction as FunctionDefinitionHandler).handler.split('.')[0];
```
Comment (smbkr) [inline plugin/bundleVisualizer.ts]: will return the value configured in `serverless.yml` for the function handler, e.g. `src/handlers/foo.main`. We use split to get just up to the `.` as the part that follows is the exported function name

File: plugin/bundleVisualizer.ts [hunk 4]
```diff
@@ -65,10 +65,23 @@ async function bundleVisualizer(this: ServerlessAnalyzeBundlePlugin): Promise<vo
...
+
+  if (!metafileName) {
+    this.serverless.cli.log(
+      `🤯 Analyze failed: function ${functionName} metadata was not found`,
+      'ServerlessAnalyzeBundlePlugin',
+      { color: 'red' },
+    );
+
+    return;
+  }
```
Comment (smbkr) [inline plugin/bundleVisualizer.ts:82]: Add explicit error handler here, since I was getting uncaught errors prior to fixing this.

**Review by adriencaccia — APPROVED** [review]
Hey @smbkr !
Thanks for the issue, the explanation and the fix in this PR :partying_face: 

I am both amazed and grateful by the quality of your contribution !

I tested your fix with both `serverless-esbuild@1.17.1` and `serverless-esbuild@1.24.1`, it works for both.

I will release your fix in a 1.0.6 version.

**Pull request merged**